Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
         self._log = log
         self._error = error
-        self._short_name = short_name
 
+        if s_name is None or l_name is None or version is None:
+            (long_name, short_name, ext_version) = self._get_extension_info()
+            self._short_name = short_name
+            self._extension_version = ext_version
+            self._log_prefix = '[%s-%s] ' % (long_name, ext_version)
+        else:
+            self._short_name = s_name
+            self._extension_version = version
+            self._log_prefix = '[%s-%s] ' % (l_name, version)
+
+    def get_extension_version(self):
+        return self._extension_version
+    
     def _get_log_prefix(self):
-        return '[%s-%s]' %(self._context._name, self._context._version)
+        return self._log_prefix if self._log_prefix else '[%s-%s] ' % (self._context._name, self._context._version)
+
+    def _get_extension_info(self):
+        if os.path.isfile(MANIFEST_XML):
+            return self._get_extension_info_manifest()
+
+        ext_dir = os.path.basename(os.getcwd())
+        (long_name, version) = ext_dir.split('-')
```

Comment:
The file is manifest.xml not HandlerEnvironment.xml, which is what I committed previously.  I'm asserting that every extension **we** ship from this repository will have manifest.xml included.

The directory structure is one that I'm less sure about, but there are existing code comments that state this is defacto.  It's our agent, so I assume the convention is what we own.  I'm not even sure I need the fallback case because we control all everything in the first place.